goprocessor/src/sawtooth_simplewallet/handlers/simplewallet_handler.go

```diff
@@ -0,0 +1,191 @@
+	}
+	debtorBalance, err := strconv.Atoi(balance)
+	if err != nil {
+		return err
+	}
+	if debtorBalance < self.getAmount() {
+		return errors.New("Transfer amount should be lesser than or equal to " + strconv.Itoa(debtorBalance))
+	}
+
+	//Get and validate balance from ledger state for creditor
+	balance = string(currentLedgerEntry[walletKeyTo])
+	if balance == "" {
+		return errors.New("Didn't find the wallet key associated with user key " + self.getUserTo())
+	}
+	creditorBalance, err := strconv.Atoi(balance)
+	if err != nil {
+		return err
+	}
+
+	// Update balance of debtor
+	updateBalance := debtorBalance - self.getAmount()
+	entry := make(map[string][]byte)
+	entry[walletKeyFrom] = []byte(strconv.Itoa(updateBalance))
+	logger.Info("Debiting balance with " + strconv.Itoa(self.getAmount()))
+	self.getContext().SetState(entry)
+
+	//Crediting to creditor
+	// Update balance of creditor
+	updateBalance = creditorBalance + self.getAmount()
+	entry = make(map[string][]byte)
+	entry[walletKeyTo] = []byte(strconv.Itoa(updateBalance))
+	logger.Info("Crediting to balance with " + strconv.Itoa(self.getAmount()))
+	self.getContext().SetState(entry)
+	return nil
+}
+
+func (self SimpleWalletHandler) withdraw() error {
+	// Get the wallet key derived from the wallet user's public key
+	walletKey := self.getNamespaceAddress() + utils.Hexdigest(self.getUserFrom())[:constants.TransactionUserAddressLength]
+	logger.Info("Got user key " + self.getUserFrom() + "wallet key " + walletKey)
+	// Get balance from ledger state
+	currentLedgerEntry, err := self.getContext().GetState([]string{walletKey})
+	if err != nil {
+		return err
+	}
+	balance := string(currentLedgerEntry[walletKey])
+	// getState() will return empty map if wallet key doesn't exist in state
+	if balance == "" {
+		return errors.New("Didn't find the wallet key associated with user key " + self.getUserFrom())
+	}
+	value, err := strconv.Atoi(balance)
+	if err != nil {
+		return err
+	}
+	if value < self.getAmount() {
+		return errors.New("Withdraw amount should be lesser than or equal to " + strconv.Itoa(value))
+	}
+	// Update balance
+	updateBalance := value - self.getAmount()
+	entry := make(map[string][]byte)
+	entry[walletKey] = []byte(strconv.Itoa(updateBalance))
+	logger.Info("Withdrawing amount: " + strconv.Itoa(self.getAmount()))
+	self.getContext().SetState(entry)
+	return nil
+}
+
+func (self SimpleWalletHandler) deposit() error {
+	// Get the wallet key derived from the wallet user's public key
+	walletKey := self.getNamespaceAddress() + utils.Hexdigest(self.getUserFrom())[:constants.TransactionUserAddressLength]
+	logger.Info("Got user key " + self.getUserFrom() + "wallet key " + walletKey)
+	// Get balance from ledger state
+	currentLedgerEntry, err := self.getContext().GetState([]string{walletKey})
+	if err != nil {
+		return err
+	}
+	balance := string(currentLedgerEntry[walletKey])
+	newBalance := 0
+	// getState() will return empty map if wallet key doesn't exist in state
```

Comment:
s/getState/GetState/

Reply:
Good catch, I will update the commented line